packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,601 @@
+              severity={severity}
+            />
+          </Link>
+        )
+      }
+    </SideNavListItemContainer>
+  );
+};
+
+export type SideNavProps = {
+  /**
+   * Side Nav header text that is usually reserved to show the App name
+   */
+  navHeaderText: string;
+  /**
+   *  Determines whether the side nav is collapsible
+   */
+  isCollapsible?: boolean;
+  /**
+   * Footer items in the side nav
+   */
+  footerItems?: SideNavFooterItem[];
+  /**
+   *  Triggers when the side nav is collapsed
+   */
+  onCollapse?(): void;
+  /**
+   * Nav items in the side nav
+   */
+  sideNavItems: SideNavItem[];
+} & Pick<HtmlProps, "testId">;
+
+const SideNav = ({
+  navHeaderText,
+  isCollapsible,
+  onCollapse,
+  sideNavItems,
+  footerItems,
+}: SideNavProps) => {
+  const [isSideNavCollapsed, setSideNavCollapsed] = useState(false);
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const processedSideNavItems = useMemo(
+    () =>
+      sideNavItems.map((item) => ({
+        ...item,
+        children: item.children?.map((childProps) => (
+          <SideNavItemContent key={childProps.id} {...childProps} />
+        )),
+      })),
+    [sideNavItems],
+  );
+
+  const sideNavCollapseHandler = useCallback(() => {
+    setSideNavCollapsed(!isSideNavCollapsed);
+    onCollapse && onCollapse();
+  }, [isSideNavCollapsed, setSideNavCollapsed, onCollapse]);
+
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        height: "100vh",
+      }}
+    >
+      <SideNavCollapsedContainer
+        tabIndex={0}
+        odysseyDesignTokens={odysseyDesignTokens}
+        isSideNavCollapsed={isSideNavCollapsed}
+        onClick={() => setSideNavCollapsed(!isSideNavCollapsed)}
+        onKeyDown={(event) => {
+          (event.key === "Enter" || event.code === "Space") &&
+            setSideNavCollapsed(!isSideNavCollapsed);
+        }}
+      >
+        <ExpandLeftIcon
+          sx={{
+            margin: `0 ${odysseyDesignTokens.Spacing1}`,
+            fontSize: "1em",
+          }}
+        />
+      </SideNavCollapsedContainer>
+      <SideNavContainer
+        odysseyDesignTokens={odysseyDesignTokens}
+        isSideNavCollapsed={isSideNavCollapsed}
+      >
+        <Box
+          sx={{
+            position: "sticky",
+            top: 0,
+          }}
+        >
+          <SideNavHeader
+            navHeaderText={navHeaderText}
+            isCollapsible={isCollapsible}
+            onCollapse={sideNavCollapseHandler}
+          />
+        </Box>
+        <Box
+          sx={{
+            flex: 1,
+            overflowY: "auto",
+          }}
+        >
+          <SideNavListContainer>
+            {processedSideNavItems?.map((item) => {
+              const {
+                id,
+                label,
+                isSectionHeader,
+                startIcon,
+                children,
+                isDefaultExpanded,
+                isDisabled,
+              } = item;
```

Comment:
Does this work now?
```suggestion
            {processedSideNavItems?.map(({
                id,
                label,
                isSectionHeader,
                startIcon,
                children,
                isDefaultExpanded,
                isDisabled,
              }) => {
```

Reply:
yes, but I also need to pass the original `item` to another component in the else block.